src/core/frames/frame_controller.ts

```diff
@@ -277,9 +289,10 @@ export class FrameController
   }
 
   private proposeVisitIfNavigatedWithAction(frame: FrameElement, element: Element, submitter?: HTMLElement) {
-    const action = getAttribute("data-turbo-action", submitter, element, frame)
+    this.action = getAttribute("data-turbo-action", submitter, element, frame) as Action
```

Comment:
Would a check with [isAction](https://github.com/hotwired/turbo/blob/main/src/core/types.ts#L3-L5) here help assure the compiler that the returned value is of type `Action`?

Reply:
https://github.com/hotwired/turbo/pull/430 extracts a [`getVisitAction` utility function](https://github.com/hotwired/turbo/pull/430/files#diff-78d8451f964182fd51330bac500ba6e71234f81aad9e8a57e682e723d0f517b3R422-R427). It might be useful here:

```typescript
export function getVisitAction(...elements: (Element|undefined)[]): Action | null {
  const action = getAttribute("data-turbo-action", ...elements)

  return isAction(action) ? action : null
}
```